src/async_wrap.cc

```diff
@@ -730,6 +734,27 @@ std::string AsyncWrap::diagnostic_name() const {
       std::to_string(static_cast<int64_t>(async_id_)) + ")";
 }
 
+Local<Object> AsyncWrap::GetOwner() {
+  return GetOwner(env(), object());
+}
+
+Local<Object> AsyncWrap::GetOwner(Environment* env, Local<Object> obj) {
+  v8::EscapableHandleScope handle_scope(env->isolate());
+  CHECK(!obj.IsEmpty());
+
+  v8::TryCatch ignore_exceptions(env->isolate());
+  while (true) {
```

Comment:
What will happen if we have a pathological case of cyclic ownership?
```
a[owner_symbol] = b;
b[owner_symbol] = a;
```

Is there a cycle safe STL algorithm for such a recursive walk?

Reply:
You could just limit the loop to an arbitrary number of iterations and crash. IMHO better than a infinite busy loop, but I have no strong  feelings about this.